src/main/java/de/zalando/aruha/nakadi/validation/EventValidation.java

```diff
@@ -51,10 +49,33 @@ private static JSONObject wrapSchemaInData(final JSONObject schema) {
         return wrapper;
     }
 
-    private static JSONObject addMetadata(final JSONObject schema) {
+    private static JSONObject addMetadata(final JSONObject schema, EventType eventType) {
         normalizeSchema(schema);
 
-        schema.getJSONObject("properties").put("metadata", new JSONObject("{\"type\": \"object\"}"));
+        final JSONObject metadata = new JSONObject();
+        final JSONObject metadataProperties = new JSONObject();
+
+        final JSONObject uuid = new JSONObject("{\"type\": \"string\", \"pattern\": \"^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$\"}");
+        final JSONObject eventTypeString = new JSONObject("{\"type\": \"string\", \"enum\": [\"" + eventType.getName() + "\"]}");
+        final JSONObject string = new JSONObject("{\"type\": \"string\"}");
+
+        // We know this is not rfc3339. We think this is enough by now
+        final String rfc3339 = "^[0-9]{4}-?[0-9]{2}-?[0-9]{2}T?[0-9]{2}:?[0-9]{2}:?[0-9]{2}(.[0-9]+)?(Z|[+-][0-9]{2}:?[0-9]{2})$";
+        final JSONObject dateTime = new JSONObject("{\"type\": \"string\", \"pattern\": \"" + rfc3339 + "\"}");
+
+        metadataProperties.put("eid", uuid);
+        metadataProperties.put("event_type", eventTypeString);
+        metadataProperties.put("occurred_at", dateTime);
+        metadataProperties.put("parent_eid", uuid);
+        metadataProperties.put("root_eid", uuid);
```

Comment:
You're right. It was based on an outdated version of the api. Removed in https://github.com/zalando/nakadi/commit/c4fe0aa7ffb3139d9abcfb387b9f83e3ebe52c66